test/harness/proxytrapshelper-overrides.js

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2016 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: allowProxyTraps helper should default throw on all the proxy trap named methods being invoked
+id: pending
+author: Jordan Harband
+includes: [proxyTrapsHelper.js]
+---*/
+
+var traps = allowProxyTraps({
+    getPrototypeOf: function () {},
+    setPrototypeOf: function () {},
+    isExtensible: function () {},
+    preventExtensions: function () {},
+    getOwnPropertyDescriptor: function () {},
+    has: function () {},
+    get: function () {},
+    set: function () {},
+    deleteProperty: function () {},
+    defineProperty: function () {},
+    enumerate: function () {},
+    ownKeys: function () {},
+    apply: function () {},
+    construct: function () {},
+});
+
+function assertTrapSucceeds(trap) {
```

Comment:
It's not enough to check that the trap didn't throw because we also want to be
sure that the specified function was invoked. If you want to enforce a loose
contract, you could assert this with a counter variable for each provided trap.
If you don't mind stricter tests (or just want to avoid all those counters),
you could

```diff
-var traps = allowProxyTraps({
-var input = {
   getPrototypeOf: function () {},
   setPrototypeOf: function () {},

 ...
   apply: function () {},
   construct: function () {},
-});
+};
+var traps = allowProxyTraps(input);
```

...and then

```diff
-function assertTrapSucceeds(trap) {
+function assertCustom(trap) {
+  if (input[trap] !== traps[trap]) {
+    throw new Test262Error('Expected custom trap');
+  }
+}
```